transit_odp/browse/tests/test_views.py

```diff
@@ -842,34 +848,172 @@ def test_operators_view_search(self, request_factory: RequestFactory):
+        )
+        # Setup a draft TXCFileAttributes
+        dataset2 = DraftDatasetFactory(organisation=org)
+        TXCFileAttributesFactory(
+            revision=dataset2.revisions.last(), service_code=all_service_codes[2]
+        )
+
+        live_revision = DatasetRevisionFactory(dataset=dataset2)
+
+        # Setup a TXCFileAttributes that will be 'Stale - 12 months old'
+        TXCFileAttributesFactory(
+            revision=live_revision,
+            service_code=all_service_codes[3],
+            operating_period_end_date=None,
+            modification_datetime=timezone.now() - datetime.timedelta(weeks=100),
+        )
+
+        # Setup a TXCFileAttributes that will be 'Stale - End Date Passed'
+        TXCFileAttributesFactory(
+            revision=live_revision,
+            service_code=all_service_codes[4],
+            operating_period_end_date=datetime.date.today()
+            - datetime.timedelta(weeks=105),
+            modification_datetime=timezone.now() - datetime.timedelta(weeks=100),
+        )
+
+        # Setup a TXCFileAttributes that will be 'Stale - OTC Variation'
+        TXCFileAttributesFactory(
+            revision=live_revision,
+            service_code=all_service_codes[5],
+            operating_period_end_date=datetime.date.today()
+            + datetime.timedelta(days=50),
+        )
+
+        # Create Seasonal Services - one in season, one out of season
+        SeasonalServiceFactory(
+            licence=bods_licence,
+            start=today,
+            end=month,
+            registration_code=int(all_service_codes[6][-1:]),
+        )
+        SeasonalServiceFactory(
+            licence=bods_licence,
+            start=month,
+            end=two_months,
+            registration_code=int(all_service_codes[7][-1:]),
         )
-        revisions = [
-            DatasetRevisionFactory(dataset=datasets[0], num_of_lines=1),
-            DatasetRevisionFactory(dataset=datasets[1], num_of_lines=2),
-        ]
 
-        PTIObservationFactory(revision=revisions[1])
+        otc_lic1 = LicenceModelFactory(number=licence_number)
+        for code in all_service_codes:
+            ServiceModelFactory(
+                licence=otc_lic1,
+                registration_number=code.replace(":", "/"),
+                effective_date=datetime.date(year=2020, month=1, day=1),
+            )
 
         request = request_factory.get("/operators/")
         request.user = UserFactory()
 
+        response = OperatorDetailView.as_view()(request, pk=org.id)
+        assert response.status_code == 200
+        context = response.context_data
+        assert context["view"].template_name == "browse/operators/operator_detail.html"
+        # One out of season seasonal service reduces in scope services to 8
+        assert context["total_in_scope_in_season_services"] == 8
+        assert context["services_require_attention_percentage"] == 75
```

Comment:
I think it's worth adding a comment to say where the 75 comes from. I think it's because there are two services that are not stale, and six that are a problem - so 6/8 (75%) are a problem.

Reply:
Good suggestion - I've added that comment here and a similar one for the other new unit test.